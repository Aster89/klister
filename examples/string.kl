#lang kernel

(example "hello world")

(example ((string-append "hello ") "world"))

(example ((string=? ((string-append "hello") " world")) "hello world"))

(example ((string=? ((string-append "hello") "world")) "hello world"))

(example (((substring 2) 3) "abcdefghijklmnop"))
(example (((substring 2) 300) "abcdefghijklmnop"))
(example (((substring 0) 2) "ab"))
(example (((substring 0) 3) "ab"))

(example (string-length "abc"))
(example (string-length ""))


(example ((string=? "a") "b"))
(example ((string=? "a") "a"))
(example ((string=? "b") "a"))
(example ((string/=? "a") "b"))
(example ((string/=? "a") "a"))
(example ((string/=? "b") "a"))
(example ((string<? "a") "b"))
(example ((string<? "a") "a"))
(example ((string<? "b") "a"))
(example ((string<=? "a") "b"))
(example ((string<=? "a") "a"))
(example ((string<=? "b") "a"))
(example ((string>? "a") "b"))
(example ((string>? "a") "a"))
(example ((string>? "b") "a"))
(example ((string>=? "a") "b"))
(example ((string>=? "a") "a"))
(example ((string>=? "b") "a"))

(example (string-upcase "aBC!"))
(example (string-upcase "Straße"))

(example (string-downcase "aBC!"))
(example (string-downcase "Straße"))
(example (string-downcase "ΚΑΟΣ"))
(example (string-downcase "Σ"))

(example (string-titlecase "aBC  twO"))
(example (string-titlecase "y2k"))
(example (string-titlecase "main straße"))
(example (string-titlecase "stra ße"))

(example (string-foldcase "aBC!"))
(example (string-foldcase "Straße"))
(example (string-foldcase "ΚΑΟΣ"))

