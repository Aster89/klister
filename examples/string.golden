"hello world" : String
"hello world" : String
(true) : Bool
(false) : Bool
"cde" : (Maybe String)
(nothing) : (Maybe String)
"ab" : (Maybe String)
(nothing) : (Maybe String)
3 : Integer
0 : Integer
(false) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
(true) : Bool
(false) : Bool
(false) : Bool
(true) : Bool
(true) : Bool
(false) : Bool
(false) : Bool
(false) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
(true) : Bool
"ABC!" : String
"STRASSE" : String
"abc!" : String
"stra\223e" : String
"\954\945\959\963" : String
"\963" : String
"Abc  Two" : String
"Y2k" : String
"Main Stra\223e" : String
"Stra Sse" : String
"abc!" : String
"strasse" : String
"\954\945\959\963" : String
